src/librustdoc/html/templates/type_layout.html

```diff
@@ -0,0 +1,45 @@
+<h2 id="layout" class="small-section-header">  {# #}
+    Layout<a href="#layout" class="anchor">§</a> {# #}
+</h2> {# #}
+<div class="docblock"> {# #}
+    {% match self.type_layout() %}
+        {% when Ok(ty_layout) %}
+            <div class="warning"> {# #}
+                <p> {# #}
+                    <strong>Note:</strong> Most layout information is <strong>completely {#+ #}
+                    unstable</strong> and may even differ between compilations. {#+ #}
+                    The only exception is types with certain <code>repr(...)</code> {#+ #}
+                    attributes. Please see the Rust Reference’s {#+ #}
+                    <a href="https://doc.rust-lang.org/reference/type-layout.html">“Type Layout”</a> {#+ #}
+                    chapter for details on type layout guarantees. {# #}
+                </p> {# #}
+            </div> {# #}
+            <p><strong>Size:</strong> {{ self.write_size(ty_layout.layout.0.borrow(), 0) | safe }}</p> {# #}
+            {% if let Some(variants) = self.variants() %}
+                <p><strong>Size for each variant:</strong></p> {# #}
+                <ul> {# #}
+                    {% for (index, layout) in variants.iter_enumerated() %}
+                        <li> {# #}
+                            <code>{{ self.variant_name(index.clone()) }}</code>: {#+ #}
+                            {{ self.write_size(layout, self.tag_size()) | safe }}
+                        </li> {# #}
+                    {% endfor %}
+                </ul> {# #}
+            {% endif %}
+        {# This kind of layout error can occur with valid code, e.g. if you try to
+           get the layout of a generic type such as `Vec<T>`. #}
+        {% when Err(LayoutError::Unknown(_)) %}
+            <p><strong>Note:</strong> Unable to compute type layout, {#+ #}
```

Comment:
I think a `<p>` should have a newline after it, though I agree with the decision to not add newlines after `<strong>` and `<code>`.